pubsub-adapter/src/main/kotlin/com/ably/pubsub/Client.kt

```diff
@@ -0,0 +1,161 @@
+   *  or forwards which orders stats from oldest to most recent. The default is backwards.
+   *  - limit (RSC6b3) - An upper limit on the number of stats returned. The default is 100, and the maximum is 1000.
+   *  - unit (RSC6b4) - minute, hour, day or month. Based on the unit selected, the given start or end times are rounded down to the start of the relevant interval depending on the unit granularity of the query.
+   *
+   * Spec: RSC6a
+   *
+   * @return A [PaginatedResult] object containing an array of [Stats] objects.
+   * @throws AblyException
+   */
+  fun stats(params: List<Param>): PaginatedResult<Stats>
+
+  /**
+   * Asynchronously queries the REST /stats API and retrieves your application's usage statistics.
+   * @param params query options:
+   *  - start (RSC6b1) - The time from which stats are retrieved, specified as milliseconds since the Unix epoch.
+   *  - end (RSC6b1) - The time until stats are retrieved, specified as milliseconds since the Unix epoch.
+   *  - direction (RSC6b2) - The order for which stats are returned in. Valid values are backwards which orders stats from most recent to oldest,
+   *  or forwards which orders stats from oldest to most recent. The default is backwards.
+   *  - limit (RSC6b3) - An upper limit on the number of stats returned. The default is 100, and the maximum is 1000.
+   *  - unit (RSC6b4) - minute, hour, day or month. Based on the unit selected, the given start or end times are rounded down to the start of the relevant interval depending on the unit granularity of the query.)
+   *
+   * Spec: RSC6a
+   *
+   * @param callback Listener which returns a [AsyncPaginatedResult] object containing an array of [Stats] objects.
+   * This callback is invoked on a background thread
+   */
+  fun statsAsync(params: List<Param>, callback: Callback<AsyncPaginatedResult<Stats>>)
+
+  /**
+   * Makes a REST request to a provided path. This is provided as a convenience
+   * for developers who wish to use REST API functionality that is either not
+   * documented or is not yet included in the public API, without having to
+   * directly handle features such as authentication, paging, fallback hosts,
+   * MsgPack and JSON support.
+   *
+   * Spec: RSC19
+   *
+   * @param method The request method to use, such as GET, POST.
+   * @param path The request path.
+   * @param params The parameters to include in the URL query of the request.
+   * The parameters depend on the endpoint being queried.
+   * See the [REST API reference](https://ably.com/docs/api/rest-api)
+   * for the available parameters of each endpoint.
+   * @param body The RequestBody of the request.
+   * @param headers Additional HTTP headers to include in the request.
+   * @return An [HttpPaginatedResponse] object returned by the HTTP request, containing an empty or JSON-encodable object.
+   */
+  fun request(
+    method: String,
```

Comment:
Perhaps worth introducing an Enum for the methods for better type safety (just for Kotlin)?